### PR TITLE
Resource endpoints audit

### PR DESCRIPTION
**Pull request by abdullah1308**
**Notes for Reviewers**

This PR fixes #

Status - https://docs.google.com/spreadsheets/d/1fPvMFjlSCo0U9sQbeMv2-NwTsZ11otDd5ZPO5bWX8Fg/edit?usp=sharing

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit 24765b16571bc18d07a7dcf79b3644eb716cc02f at: https://649735c13a586408ab82c3ca--meshery-docs-preview.netlify.app

**Comment by abdullah1308** [comment]
@Lee Meshkit release required to get access to counts for `/api/meshmodels/categories` and `/api/meshmodels/categories/{category}` - https://github.com/meshery/meshkit/pull/323

**Comment by l5io** [comment]
🚀 Preview for commit e6b176f52a82745d991ff949150091c0a99ba51a at: https://6497632ce3d5791fc8f94b01--meshery-docs-preview.netlify.app

**Comment by abdullah1308** [comment]
`/api/integrations/credentials` will be updated after #7634 is merged

**Comment by l5io** [comment]
🚀 Preview for commit 4cc88635b60c3b8fe62a65d749b0a53c234e3dc0 at: https://649776c7b48fbc2ad2584a66--meshery-docs-preview.netlify.app

**Comment by leecalcote** [comment]
@alphaX86 there is a collection of changes around the performance APIs (and others) that may affect mesheryctl. Do you mind taking a look?

**Comment by abdullah1308** [comment]
TODO: Meshmap changes

File: server/internal/graphql/model/models_gen.go [hunk 1]
```diff
@@ -69,8 +69,10 @@ type CatalogPattern struct {
 }
 
 type CatalogSelector struct {
-	Search string `json:"search"`
-	Order  string `json:"order"`
+	Page     string `json:"page"`
+	PageSize string `json:"pageSize"`
```
Comment (MUzairS15) [inline server/internal/graphql/model/models_gen.go:73]: Let's be consistent with `json` tags as well. `page_size`. You can update GraphQL schema in a way that this tag is generated appropriately.

**Review by MUzairS15 — COMMENTED** [review]
For [this](https://github.com/meshery/meshery/pull/7917/files#diff-7171fb49b576deb621a891f27692b78822c2c459e5e58117e7cf7b462040dad0R481) to be reflected in UI, you need to re-run relay scripts and UI changes if any.

**Comment by MUzairS15** [comment]
Nice work @abdullah1308 , consistency is key. 👏🏻

**Comment by leecalcote** [comment]
> Nice work @abdullah1308 , consistency is key. 👏🏻

Ditto (on both points). 😄

File: server/internal/graphql/model/models_gen.go (hunk 1 above)
Comment (abdullah1308) [reply server/internal/graphql/model/models_gen.go]: @MUzairS15 done

Comment (abdullah1308) [reply server/internal/graphql/model/models_gen.go]: @MUzairS15 What do you mean by relay scripts ? Can you point them out to me please.

**Comment by abdullah1308** [comment]
> TODO: Meshmap changes

@leecalcote @MUzairS15 Meshmap changes done

**Comment by l5io** [comment]
🚀 Preview for commit 95d3888fdbf243c5e11dca23bb5cece4a3927868 at: https://6499f17116b5790d01f38613--meshery-docs-preview.netlify.app

File: server/internal/graphql/model/models_gen.go (hunk 1 above)
Comment (MUzairS15) [reply server/internal/graphql/model/models_gen.go]: https://github.com/meshery/meshery/blob/cc6073c2a8e1aeb5d94e7940aa6422f4409978d5/ui/components/graphql
Run really script as follows: `cd ui; npm run relay;`

Comment (abdullah1308) [reply server/internal/graphql/model/models_gen.go]: ran relay script

**Comment by leecalcote** [comment]
@MUzairS15 a merge conflict has come up. @abdullah1308 will be out for the next couple days. Will you carry this forward?

**Comment by leecalcote** [comment]
@MUzairS15 @theBeginner86 @Abhishek-kumar09 Please do coordinate the PRs related to this across the components.

**Pull request merged**